src/ValidatorSubstitute.php

```diff
@@ -0,0 +1,239 @@
+        }
+        return $this->fields['users_id'] == Session::getLoginUserID();
+    }
+
+    public function canCreateItem()
+    {
+        if (!isset($this->fields['users_id'])) {
+            return false;
+        }
+
+        return $this->fields['users_id'] == Session::getLoginUserID();
+    }
+
+    public function canUpdateItem()
+    {
+        if (!isset($this->fields['users_id'])) {
+            return false;
+        }
+        return $this->fields['users_id'] == Session::getLoginUserID();
+    }
+
+    public function canDeleteItem()
+    {
+        if (!isset($this->fields['users_id'])) {
+            return false;
+        }
+        return $this->fields['users_id'] == Session::getLoginUserID();
+    }
+
+    public function canPurgeItem()
+    {
+        if (!isset($this->fields['users_id'])) {
+            return false;
+        }
+        return $this->fields['users_id'] == Session::getLoginUserID();
+    }
+
+    public function showForUser(User $user): bool
+    {
+        if ($user->isNewItem()) {
+            return false;
+        }
+
+        $can_edit = ($user->fields['id'] == Session::getLoginUserID());
+        if (!$can_edit) {
+            return false;
+        }
+
+        TemplateRenderer::getInstance()->display('pages/admin/user.substitute.html.twig', [
+            'item'        => $this,
+            'user'        => $user,
+            'substitutes' => $user->getSubstitutes(),
+            'delegators'  => $user->getDelegators(),
+            'canedit'     => $can_edit,
+        ]);
+
+        return true;
+    }
+
+    public function prepareInputForUpdate($input)
+    {
+        if (isset($input['users_id']) && $input['users_id'] != $this->fields['users_id']) {
+            // Do not change the user.
+            Session::addMessageAfterRedirect(__('Cannot change the validation delegator'));
+            return [];
+        }
+
+        return $input;
+    }
+
+    /**
+     * Update the substitutes list for the current user
+     *
+     * @param array $input
+     * @return boolean
+     */
+    public function updateSubstitutes($input): bool
+    {
+        $validator_substitute = new self();
+
+        // The user to be substituted
+        if ($input['users_id'] != Session::getLoginUserID()) {
+            Session::addMessageAfterRedirect(__('You cannot change substitutes for this user.'), true, ERROR);
+            return false;
+        }
+        $user = User::getById($input['users_id']);
+        if (!($user instanceof User)) {
+            return false;
+        }
+
+        // Get the old substitutes list
+        $old_substitutes = $user->getSubstitutes();
+
+        // Store the overall sucess of the changes below
+        $success = true;
+
+        if (isset($input['substitutes'])) {
+            if (in_array($input['users_id'], $input['substitutes'])) {
+                Session::addMessageAfterRedirect(__('Cannot add a user as substitute of himself.'), true, ERROR);
```

Comment:
```suggestion
                Session::addMessageAfterRedirect(__('A user cannot be their own substitute.'), true, ERROR);
```
Gender neutral language should be used.
"Cannot add a user as substitute of themself." could be acceptable as well but using "themself" isn't very common anymore and the plural form "themselves" wouldn't be correct here.